### PR TITLE
KTOR-1965 Add an option to ignore trailing slash in routing

### PR DESCRIPTION
**Pull request by rsinukov**
**Subsystem**
Server, routing

**Motivation**
https://youtrack.jetbrains.com/issue/KTOR-1165

**Solution**
Add feature to ignore trailing slash in routing

### CODE REVIEW

**Comment by cy6erGn0m** [comment]
Not sure if `isTrailingSlashMatters` a good name. I personally would, in opposite, name it `ignoreTrailingSlash`. But I don't insist

File: ktor-server/ktor-server-core/api/ktor-server-core.api [hunk 1]
```diff
@@ -1440,6 +1440,7 @@ public final class io/ktor/routing/Routing : io/ktor/routing/Route {
 	public fun <init> (Lio/ktor/application/Application;)V
 	public final fun getApplication ()Lio/ktor/application/Application;
 	public final fun interceptor (Lio/ktor/util/pipeline/PipelineContext;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
+	public final fun isTrailingSlashMatters ()Z
```
Comment (e5l) [inline ktor-server/ktor-server-core/api/ktor-server-core.api:1443]: We need a different name for sure

**Review by e5l — CHANGES_REQUESTED** [review]
Let's brainstorm on the parameter name. I would also check the test coverage of the routing feature

**Comment by e5l** [comment]
Agree with @cy6erGn0m, `ignoreTrailingSlash` is good. We also can try `allowTrailingSlashHandlers`

**Comment by rsinukov** [comment]
i like @cy6erGn0m  suggestion of `ignoreTrailingSlash `

File: ktor-server/ktor-server-core/jvm/src/io/ktor/routing/IgnoreTrailingSlash.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+import io.ktor.application.*
+import io.ktor.util.*
+
+private val IgnoreTrailingSlashAttributeKey: AttributeKey<Unit> = AttributeKey("IgnoreTrailingSlashAttributeKey")
+
+internal var ApplicationCall.ignoreTrailingSlash: Boolean
+    get() = attributes.contains(IgnoreTrailingSlashAttributeKey)
+    private set(value) = when (value) {
+        true -> attributes.put(IgnoreTrailingSlashAttributeKey, Unit)
```
Comment (e5l) [inline ktor-server/ktor-server-core/jvm/src/io/ktor/routing/IgnoreTrailingSlash.kt:15]: why not `if else`?

**Review by e5l — APPROVED** [review]

**Pull request merged**